hnn_core/viz.py

```diff
@@ -465,7 +465,10 @@ def plot_spikes_hist(cell_response, trial_idx=None, ax=None, spike_types=None,
     elif isinstance(color, list):
         color_cycle = cycle(color)
 
-    bins = np.linspace(0, spike_times[-1], 50)
+    if len(cell_response.times) > 0:
+        bins = np.linspace(0, cell_response.times[-1], 50)
+    else:
+        bins = np.linspace(0, spike_times[-1], 50)
```

Comment:
I have explored this solution before, but I found that it reduces temporal resolution if there are events early on in the trial. Current implementation (top figure) shows 3 bins, at times of APs, this implementation shows 1 bin (bottom figure). I agree that this one makes more sense for the histogram and I think it also make sense to show spiking of the pyramidal neurons and basket cells. But we lose resolution when showing our proximal and distal drives (so they might just be lumped together into one bin and you can't infer the standard deviation of the spikes).

![current_bins](https://github.com/user-attachments/assets/339150e9-2f40-4949-874b-66d2cfa90cfb)

![new_bins](https://github.com/user-attachments/assets/acc85431-cbde-4cdf-8a23-7192f8186e28)

Reply:
Personally my preference is to have the bin resolution tied to the length of the simulation, Since we have an option of raster plots it's possible to see everything with the "maximal" temporal resolution.

However the point is well taken that there should be a `bins` argument to this!